### PR TITLE
Replace `QRegExp` with `QRegularExpression`

### PR DESCRIPTION
**Pull request by w0xlt**
Picking up https://github.com/bitcoin-core/gui/pull/606 (labeled "Up for grabs") and applying https://github.com/bitcoin-core/gui/pull/606#pullrequestreview-984607067 and https://github.com/bitcoin-core/gui/pull/606#issuecomment-1137149907.

Replaces occurrences of `QRegExp` usage with `QRegularExpression` as part of the roadmap for Qt6 integration.

Fixes https://github.com/bitcoin-core/gui/issues/578

### CODE REVIEW

**Comment by hebasto** [comment]
@w0xlt 

Thank you for picking it up!

What is your opinion about https://github.com/bitcoin-core/gui/pull/606#issuecomment-1138795241?

**Comment by w0xlt** [comment]
> What is your opinion about https://github.com/bitcoin-core/gui/pull/606#issuecomment-1138795241?

That looks good to me. I'll take a look at it.
Would you suggest a class/file to allocate the utility function?

**Comment by hebasto** [comment]
> > What is your opinion about [#606 (comment)](https://github.com/bitcoin-core/gui/pull/606#issuecomment-1138795241)?
> 
> That looks good to me. I'll take a look at it. Would you suggest a class/file to allocate the utility function?

`namespace GUIUtil` in `qt/guiutil.{h,cpp}`?

File: src/qt/test/optiontests.cpp [hunk 1]
```diff
@@ -122,3 +123,19 @@ void OptionTests::parametersInteraction()
...
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "psbt");
+
+    filter = QString("Comma separated file (*.csv)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "csv");
+
+    filter = QString("PNG Image (*.png)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "png");
+
+    filter = QString("Wallet Data (*.dat)");
+    QCOMPARE(GUIUtil::extractFirstSuffixFromFilter(filter), "dat");
```
Comment (hebasto) [inline src/qt/test/optiontests.cpp:139]: Why 4 tests with the only suffix? Maybe just two tests? First test with one suffix, another one with two suffixes.

File: src/qt/test/optiontests.cpp [hunk 2]
```diff
@@ -122,3 +123,19 @@ void OptionTests::parametersInteraction()
...
+
```
Comment (hebasto) [inline src/qt/test/optiontests.cpp:140]: Drop empty line?

File: src/qt/utilitydialog.cpp [hunk 3]
```diff
@@ -22,7 +22,7 @@
 #include <QCloseEvent>
 #include <QLabel>
 #include <QMainWindow>
-#include <QRegExp>
+#include <QRegularExpression>
```
Comment (hebasto) [inline src/qt/utilitydialog.cpp:25]: Add `#include <QString>` as well (due to using `QStringLiteral`)?

**Review by hebasto — COMMENTED** [review]
Approach ACK c0c8b33748637372ee5b267f6bbcce33a5380f4d.

Suggesting to split the first commit into separated ones:
- factoring out the `extractFirstSuffixFromFilter9()` function
- replacement `QRegExp` with `QRegularExpression`

Also please apply [`clang-format-diff.py`](https://github.com/bitcoin/bitcoin/tree/master/contrib/devtools#clang-format-diffpy) for each your commit.

File: src/qt/guiutil.h [hunk 4]
```diff
@@ -123,6 +123,13 @@ namespace GUIUtil
      */
     QString getDefaultDataDirectory();
 
+
+    /** Extract first suffix from filter pattern "Description (*.foo)" or "Description (*.foo *.bar ...).
+
+      @param[in] filter Filter specification such as "Comma Separated Files (*.csv)"
+    */
```
Comment (hebasto) [inline src/qt/guiutil.h:130]: Please consult with [Coding Style (Doxygen-compatible comments)](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-doxygen-compatible-comments).

File: src/qt/test/optiontests.cpp (hunk 2 above)
Comment (w0xlt) [reply src/qt/test/optiontests.cpp:131]: Done in https://github.com/bitcoin-core/gui/commit/67364ebe4c499eb8effe8dac11a5e3648f30c6c7.

File: src/qt/utilitydialog.cpp (hunk 3 above)
Comment (w0xlt) [reply src/qt/utilitydialog.cpp:25]: Done in https://github.com/bitcoin-core/gui/pull/620/commits/ace9af5688662a56d855bd26f36bc5c145b909e3.

File: src/qt/guiutil.h (hunk 4 above)
Comment (w0xlt) [reply src/qt/guiutil.h]: Done in https://github.com/bitcoin-core/gui/pull/620/commits/c378535e28e0c54353fde9c4a66cb3134a3ee742.

**Comment by w0xlt** [comment]
@hebasto thanks for the review.
Force-pushed https://github.com/bitcoin-core/gui/pull/620/commits/67364ebe4c499eb8effe8dac11a5e3648f30c6c7 addressing your suggestions.

**Review by hebasto — APPROVED** [review]
ACK 67364ebe4c499eb8effe8dac11a5e3648f30c6c7

**Comment by laanwj** [comment]
Code review and lightly tested ACK 7e4ae5ff38e149f469369887858e59c79bc332ae

**Pull request merged**